qpsolvers_benchmark/benchmark.py

```diff
@@ -175,43 +183,63 @@ def parse_command_line_arguments():
 
 
 def find_results_file(args):
-    if args.command in ["check_results", "report"]:
+    """
+    Find the path to the results file.
+
+    Args:
+        args (argparse.Namespace): The arguments passed in the command line.
+
+    Raises:
+        FileNotFoundError: If the file was not found.
+
+    Returns:
+        str: The path to the results file.
+    """
+    if args.command in ["check_results", "report", "plot"]:
         results_file = (
-            args.results_file
-            if args.results_file
-            else f"results/{args.test_set}.csv"
-        )
+            os.path.abspath(args.results_file) if args.results_file
+            else os.path.join(os.path.abspath(os.path.dirname(args.test_set_path)),
+                      "results",
+                      os.path.split(args.test_set_path)[1].replace(".py", ".csv")
+                            )
+                        )
         if not os.path.exists(results_file):
             raise FileNotFoundError(f"results file '{results_file}' not found")
     else:
-        results_dir = os.path.join(os.path.dirname(__file__), "results")
-        results_file = os.path.join(results_dir, f"{args.test_set}.csv")
+        if args.command == "run" and args.results_path:
+            results_dir = os.path.abspath(args.results_path)
+        else:    
+            testset_dir = os.path.dirname(args.test_set_path)
+            results_dir = os.path.join(testset_dir, "results")
+        if not os.path.exists(results_dir):
+            os.mkdir(results_dir)
+        results_file = os.path.join(results_dir, os.path.split(args.test_set_path)[1].replace(".py", ".csv"))
     return results_file
+       
 
-
-def load_test_set(name: str) -> TestSet:
+def load_test_set(path : str) -> TestSet:
     """
     Load a test set.
-
+    
     Args:
-        name: Name of the test set.
+        path (str): path to the .py file containing the class definition of the TestSet
```

Comment:
```suggestion
        path: path to the .py file containing the class definition of the TestSet
```
Sphinx will derive it from type annotations.